src/main/java/core/basesyntax/db/Storage.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.db;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Storage {
+    private static final Map<String, Integer> fruitStorage = new HashMap<>();
+
+    public static void setBalance(String fruit, int quantity) {
+        if (quantity < 0) {
+            throw new RuntimeException("quantity can't be negative");
+        }
+        fruitStorage.put(fruit, quantity);
+    }
+
+    public static void addFruit(String fruit, int quantity) {
+        if (quantity < 0) {
+            throw new RuntimeException("quantity can't be negative");
+        }
+        fruitStorage.put(fruit, fruitStorage.getOrDefault(fruit, 0) + quantity);
+    }
+
+    public static void removeFruit(String fruit, int quantity) {
+        if (quantity < 0) {
+            throw new RuntimeException("quantity can't be negative");
+        }
+        fruitStorage.put(fruit, fruitStorage.getOrDefault(fruit, 0) - quantity);
```

Comment:
According to the checklist, handling the Purchase operation should include a check for a positive balance before saving it in the Storage. The current implementation of `removeFruit` method could lead to negative values in the storage, which violates this requirement. Make sure to add a check to ensure that the resulting balance after removal is not negative, and throw a `RuntimeException` if the balance would go negative.